File: python/conjecture.py
#!/usr/bin/env python
# File: conjecture.py
# Name: D.Saravanan
# Date: 13/07/2021

""" Script for 3n + 1 collatz conjecture """


def conjecture(n: int) -> list[int]:
    """collatz conjecture"""
    sequence: list[int] = [n]

    while n != 1:
        n = n // 2 if n % 2 == 0 else (3 * n) + 1
        sequence = sequence + [n]

    return sequence


if __name__ == "__main__":
    print(conjecture(27))

File: julia/conjecture.jl
#!/usr/bin/env julia
# File: conjecture.jl
# Name: D.Saravanan
# Date: 06/11/2023

""" Script for 3n + 1 collatz conjecture """


function conjecture(n::Int)::Vector{Int}
    """collatz conjecture"""
    sequence = [n]

    while n != 1
        n = n % 2 == 0 ? fld(n, 2) : (3 * n) + 1
        #append!(sequence, n)
        sequence = vcat(sequence, [n])
    end

    return sequence
end


function main()
    println(conjecture(27))
end

main()
